src/Repository/MetadataStatementRepository.php

```diff
@@ -68,18 +70,12 @@ public function __construct(
         private readonly string $jwtMdsRootCertFileName,
         private readonly string $mdsCacheDir,
         private readonly CertificateChainValidator $certificateChainValidator,
+        private readonly SerializerInterface $serializer,
     ) {
         $payload = $this->warmCache();
-        $data = json_decode($payload, true, flags: JSON_THROW_ON_ERROR);
-
-        if (!is_array($data)) {
-            throw new RuntimeException('Unable to read the contents from the JWT metadata statement service file');
-        }
-
-        /** @var array<string, mixed> $datum */
-        foreach ($data['entries'] as $datum) {
-            $entry = MetadataBLOBPayloadEntry::createFromArray($datum);
 
+        $blob = $this->serializer->deserialize($payload, MetadataBLOBPayload::class, JsonEncoder::FORMAT);
```

Comment:
Is this just another way of decoding json?

Edit: I see this is the way the webauthn lib works :+1: